core/model/src/main/java/org/eclipse/rdf4j/model/impl/SimpleIRI.java

```diff
@@ -102,4 +102,13 @@ public String getLocalName() {
 		return iriString.substring(localNameIdx);
 	}
 
+	//conversion function to decode percent-encoded chars, Issue: 1291
+	@Override
+	public IRI percentageDecode(IRI url){
+		String url_ = url.toString();
+		url_ = url_.replaceAll("%", "");
+		url = Values.iri(url_);
+        return url;
+	}
```

Comment:
Couple of things: first of all, I don't think `IRI` itself (or the `SimpleIRI` implementation) is really the best place for this. What we're looking for in issue #1291 is probably more a SPARQL function, or perhaps a custom option in the Rio parsers to use percent-decoding.

Second: this function as you currently have it doesn't percent-decode. All it does is remove any %-chars from the IRI - but that's not decoding it. 

I would suggest that we continue discussion on the issue ticket itself, to talk a bit more about the best solution, if you're still interested in working on this.